### PR TITLE
Testing error cases

### LINKED ISSUE #997: Testing page for new documentation
## Basic Info

* Faraday Version: master
* Ruby Version: 2.6.3

## Issue description

There is no documentation on testing with Faraday. Where I work Ruby on Rails was introduced because it's cool. Little thought was given to testing red-cases. I'd like to contribute some documentation to help others

## Steps to reproduce

* visit website.
* Click `usage`
* follow footer next links until you can proceed no further right (https://lostisland.github.io/faraday/usage/streaming)
* notice no testing in usage :scream_cat:

### PR DESCRIPTION
**Pull request by Lewiscowles1986**
Initial attempt at documenting testing error cases

## Description

Adds to documentation an example about testing errors.

## Todos

List any remaining work that needs to be done, i.e:
- [x] Tests
- [x] Documentation

## Additional Notes

# Fixes #997

### CODE REVIEW

File: docs/usage/testing.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+layout: documentation
+title: "Testing error cases"
```
[the remark this answers is not in the document]
Comment (Lewiscowles1986) [reply docs/usage/testing.md]: It is a repo you have more experience with.

I can assure you long comprehensive articles are one way to ensure things are not read. I'm stating this as a former lecturer and life-long learner. The longer the source text, the fewer people will read it, or worse... Skim read.

Perhaps this is the wrong place, or we might pair on this at some point at work.

File: docs/usage/testing.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  let(:url) { "http://some-url-to.visit" }
+  let(:connection) { class_double(Faraday) }
+
+  before do
+    allow(Faraday).to receive(:new).with(url: url).and_return(connection)
+  end
+
+  # checks for http status 2XX or similar
+end
+```
```
[the remark this answers is not in the document]
Comment (Lewiscowles1986) [reply docs/usage/testing.md]: I would love to know more about this middleware.

I would like to have production knowledge of it's use before I attempt to teach others about it.

File: docs/usage/testing.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+class SomeHttpService
+  class Error < StandardError; end
+
+  def request(url, method)
+    connection.public_send(method) do |req|
+      req.url(url)
+    end
+  end
+end
+```
```
[the remark this answers is not in the document]
Comment (Lewiscowles1986) [reply docs/usage/testing.md]: I apologize. I will try to look at some code and come up with something cleaner. Please note I will need to rewrite tests if this changes to reflect we are not calling public_send. This is fine, but it becomes verb specific and I'll have an urge to make table tests, that you may not like.

File: docs/usage/testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (olleolleolle) [reply docs/usage/testing.md]: The RaiseError Middleware is documented: https://lostisland.github.io/faraday/middleware/raise-error 

But the source code is almost the same length. https://github.com/lostisland/faraday/blob/master/lib/faraday/response/raise_error.rb

**Comment by Lewiscowles1986** [comment]
Should I close this in favour of #1000, or block until #1000 is merged, and re-approach, perhaps making the article less about the code, and more about the intent?

**Pull request closed without merging**